pkg/cfn/manager/waiters.go

```diff
@@ -107,6 +115,43 @@ func (c *StackCollection) waitWithAcceptors(i *Stack, acceptors []request.Waiter
 	return nil
 }
 
+func (c *StackCollection) waitWithAcceptorsChangeSet(i *Stack, changesetName *string, acceptors []request.WaiterAcceptor) error {
+	desiredStatus := fmt.Sprintf("%v", acceptors[0].Expected)
+	msg := fmt.Sprintf("waiting for CloudFormation changeset %q for stack %q to reach %q status", *changesetName, *i.StackName, desiredStatus)
+	ctx, cancel := context.WithTimeout(context.Background(), c.spec.WaitTimeout)
+	defer cancel()
+	startTime := time.Now()
+	w := request.Waiter{
+		Name:        strings.Join([]string{"waitCS", *i.StackName, *changesetName, desiredStatus}, "_"),
+		MaxAttempts: 1024, // we use context deadline instead
+		Delay:       makeWaiterDelay(),
+		Acceptors:   acceptors,
+		NewRequest: func(_ []request.Option) (*request.Request, error) {
+			input := &cfn.DescribeChangeSetInput{
+				StackName:     i.StackName,
+				ChangeSetName: changesetName,
+			}
+			logger.Debug(msg)
+			req, _ := c.cfn.DescribeChangeSetRequest(input)
+			req.SetContext(ctx)
+			return req, nil
+		},
+	}
+	logger.Debug("start %s", msg)
+	if waitErr := w.WaitWithContext(ctx); waitErr != nil {
+		s, err := c.describeStackChangeset(i, changesetName)
+		if err != nil {
+			logger.Debug("describeChangesetErr=%v", err)
+		} else {
+			logger.Critical("unexpected status %q while %s", *s.Status, msg)
+			c.troubleshootStackFailureCause(i, desiredStatus)
```

Comment:
Ok, so this is being used in wait-for-update code path, but there is still a chance of error when creating a changeset that we don't have a way to troubleshoot like this just yet?

Reply:
Yes there is a chance that changeset cannot be created. The changeset doesn't have events as far as i can see. We could take out the `troubleshootStackFailureCause` and actually replace with this from describing the changeset:
```
	// A description of the change set's status. For example, if your attempt to
	// create a change set failed, AWS CloudFormation shows the error message.
	StatusReason *string `type:"string"`
```
Do you think that would be better?

Reply:
I think `troubleshootStackFailureCause` makes sense for the execution of the changeset, as this is basically like `create cluster` except that its success events are different.

But for the creation of the changeset then it doesn't fit really, so i've added code to log the reason returned

Reply:
sgtm